src/test/util/net.cpp

```diff
@@ -22,8 +22,8 @@ void ConnmanTestMsg::NodeReceiveMsgBytes(CNode& node, Span<const uint8_t> msg_by
             nSizeAdded += it->m_raw_message_size;
         }
         {
-            LOCK(node.cs_vProcessMsg);
-            node.vProcessMsg.splice(node.vProcessMsg.end(), node.vRecvMsg, node.vRecvMsg.begin(), it);
+            LOCK(node.m_process_msgs_mutex);
```

Comment:
Can we add `AssertLockNotHeld` above this line?

```suggestion
            AssertLockNotHeld(node.m_process_msgs_mutex);
            LOCK(node.m_process_msgs_mutex);
```